### PR TITLE
Scala CLI candidate and initial version

### PR DESCRIPTION
**Pull request by lwronski**

### CODE REVIEW

**Comment by helpermethod** [comment]
Hi @lwronski!

I fear that the Scala CLI distribution is not packaged as an SDK and thus not suitable to be added as a candidate to SDKMAN!.

SDKMAN! requires that a distribution has a top-level directory after extraction, and a bin folder directly underneath it, something like

image

The Scala CLI distribution on the other hand looks like this

image

**Comment by lwronski** [comment]
@helpermethod Thanks for reviewing, I want to prepare packages that will be supported by SKDMAN. Can you confirm that this is the correct structure for SDKMAN?

```
$ unzip -l scala-cli-x86_64-pc-linux-static-sdk.zip 
Archive:  scala-cli-x86_64-pc-linux-static-sdk.zip
  Length      Date    Time    Name
---------  ---------- -----   ----
        0  02-02-2022 19:24   scala-cli-x86_64-pc-linux-static-sdk/
        0  02-02-2022 19:24   scala-cli-x86_64-pc-linux-static-sdk/bin/
  4600137  02-02-2022 19:24   scala-cli-x86_64-pc-linux-static-sdk/bin/scala-cli
---------                     -------
  4600137                     3 files
$ unzip -l  scala-cli-x86_64-apple-darwin-sdk.zip 
Archive:  scala-cli-x86_64-apple-darwin-sdk.zip
  Length      Date    Time    Name
---------  ---------- -----   ----
        0  02-02-2022 19:36   scala-cli-x86_64-apple-darwin-sdk/
        0  02-02-2022 19:36   scala-cli-x86_64-apple-darwin-sdk/bin/
  4599844  02-02-2022 19:36   scala-cli-x86_64-apple-darwin-sdk/bin/scala-cli
---------                     -------
  4599844                     3 files
$ unzip -l scala-cli-x86_64-pc-win32-sdk.zip 
Archive:  scala-cli-x86_64-pc-win32-sdk.zip
  Length      Date    Time    Name
---------  ---------- -----   ----
        0  02-02-2022 18:26   scala-cli-x86_64-pc-win32-sdk/
        0  02-02-2022 18:26   scala-cli-x86_64-pc-win32-sdk/bin/
  4602380  02-02-2022 18:26   scala-cli-x86_64-pc-win32-sdk/bin/scala-cli.exe
---------                     -------
  4602380                     3 files
```

**Comment by helpermethod** [comment]
Hi @lwronski, 

yes, this is the correct structure. Thanks for putting in the time! `scala-cli` would be a great addition to SDKMAN!, I'm a big fan of Scala CLI myself!

BTW, did you know that you can also publish releases as a Vendor via the Vendor API?

https://sdkman.io/vendors

**Comment by lwronski** [comment]
I added generating packages to scala-cli which are supported by SDKMAN, so after the next release I will update `ScalaCliMigrations`.  The next version of Scala CLI should be released in the next week. 

@helpermethod I know about publishing via Vendor API. I thought to do it manually the first time and then I would like to automate it using the Vendor API.

**Comment by lwronski** [comment]
From version `0.1.1`,  Scala CLI publishes packages such as `scala-cli-x86_64-apple-darwin-sdk.zip`, `scala-cli-x86_64-pc-linux-static-sdk.zip` and `cala-cli-x86_64-pc-win32-sdk.zip`. So I updated ScalaCLIMagrations with adding this packages to Migrations.

File: src/main/scala/io/sdkman/changelogs/ScalaCliMigrations.scala [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+@ChangeLog(order = "070")
+class ScalaCliMigrations {
+  @ChangeSet(
+    order = "001",
+    id = "001_add_scala-cli_candidate",
+    author = "lwronski"
+  )
+  def migration001(implicit db: MongoDatabase) =
+    Candidate(
+      candidate = "scala-cli",
```
[the remark this answers is not in the document]
Comment (lwronski) [reply src/main/scala/io/sdkman/changelogs/ScalaCliMigrations.scala]: Sure, so I set name of `candidate` to `scalacli`.

**Review by marc0der — APPROVED** [review]
LGTM!

**Pull request merged**